### PR TITLE
Node 18+ support updates

### PR DESCRIPTION
**Pull request by blackfalcon**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #189, #204, #130, #208, #209, #150, #206, #160, #156

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

BREAKING CHANGE: This commit removes all legacy es6 build processes including light-dom support, fixed CSS, non-CSS variable support, and es5 bundling.

BREAKING CHANGE: This commit will remove all support for ID attribute reflection from the parent element to the inner button element.

BREAKING CHANGE: This commit will remove all legacy support for the placement of icons within the element other than supporting icons as slotted content.

feat(ondark): update background values #156
This commit addresses an issue related to the updated color spec for the button when placed on dark backgrounds.

feat(enum): add variants support #160
This update deprecates the use of secondary and tertiary UI as individual attributes and moves them to the appropriate enum attribute of `variant`.

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Review by jordanjones243 — APPROVED** [review]
This is approved, but might be beneficial for @jason-capsule42 to review as well, as this is a large amount of files being changed and it would adhere to our new working agreement.

File: demo/index.html [hunk 1]
```diff
@@ -3,13 +3,15 @@
...
+    <title>Auro Web Component Generator | auro-button custom element</title>
     <link
       rel="stylesheet"
       type="text/css"
       href="https://unpkg.com/prismjs@latest/themes/prism.css"
     />
-    <link rel="stylesheet" type="text/css" href="css/style.css" />
+    <link rel="stylesheet" href="https://unpkg.com/@alaskaairux/design-tokens@latest/dist/tokens/CSSCustomProperties.css">
+    <link rel="stylesheet" type="text/css" href="https://unpkg.com/@alaskaairux/webcorestylesheets@latest/dist/demoWrapper.css" />
+    <link rel="stylesheet" type="text/css" href="https://unpkg.com/@alaskaairux/webcorestylesheets@latest/dist/elementDemoStyles.css" />
```
Comment (jason-capsule42) [inline demo/index.html:14]: These should be JSDeliver rather than unpkg

**Comment by jason-capsule42** [comment]
The `Dos and Don'ts` section seems to have changed from the docsite. Is this intended?

Screenshot 2023-05-31 at 7 58 10 AM

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: demo/index.html (hunk 1 above)
Comment (blackfalcon) [reply demo/index.html]: Addressed.

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**